### ISSUE TITLE
Question: does this extension supports `.config` folder?

### ISSUE THREAD

**Issue by gian1200** [microsoft/vscode-eslint#1932]
I currently have a `eslint.config.mjs` file in the root of a project.

In order to organize all config files, I was thinking about moving them to a `.config` folder, inside the project.

Does `vscode-eslint` support reading ESLint config file from inside `.config`, or only from the root?

ESLint does support it (with manual configuration), but am not sure if the extension does.

**Comment by dbaeumer** [comment]
The extension does support it as well but also config. The setting is `eslint.options`

**Comment by gian1200** [comment]
Thanks for the answer. However, when setting the following configuration, I'm a little bit puzzled by the logs.

- `.vscode/settings.json`:
```json
{
	"eslint.validate": [
		"javascript",
		"html"
	],
	"eslint.options": {
		"overrideConfigFile": "aaa/eslint.config.mjs" <-- invalid path
	}
}
```
- Log:
```sh
Error: ENOENT: no such file or directory, stat '<project-path>\aaa\eslint.config.mjs'
[Error - 3:10:55 PM] An unexpected error occurred:
[Error - 3:10:55 PM] Error: ENOENT: no such file or directory, stat '<project-path>\.config\aaa\eslint.config.mjs'
[Error - 3:10:55 PM] An unexpected error occurred:
[Error - 3:10:55 PM] Error: ENOENT: no such file or directory, stat '<project-path>\.config\aaa\eslint.config.mjs'
```

It seems that `.config` is supported by default (workspace root is evaluated first, then `.config` folder). However, When setting the correct path, I'm still getting an error:

- `.vscode/settings.json`:
```json
{
	"eslint.validate": [
		"javascript",
		"html"
	],
	"eslint.options": {
		"overrideConfigFile": ".config/eslint.config.mjs" <-- valid path
	}
}
```

- Log:
```
[Error - 3:15:52 PM] An unexpected error occurred:
[Error - 3:15:52 PM] Error: ENOENT: no such file or directory, stat '<project-path>\.config\.config\eslint.config.mjs'
```

**Comment by dbaeumer** [comment]
What happens if you override the config file on the command line in the terminal ?

**Comment by gian1200** [comment]
Do you mean to run `eslint . --config .config/eslint.config.mjs`? It works fine

```shell
npx eslint . --config .config/eslint.config.mjs <-- works fine
```

**Comment by dbaeumer** [comment]
Can you please provide me with a GitHub repository that I can clone that has a minimal setup that demos this. Would help to understand why this fails. Not sure if this is a ESLint or extension problem.

**Comment by gian1200** [comment]
Done!

https://github.com/gian1200/reproduce-vscode-eslint-issue

### Output

```shell
[Info  - 6:50:33 PM] ESLint server is starting.
[Info  - 6:50:33 PM] ESLint server running in node v20.15.1
[Info  - 6:50:33 PM] ESLint server is running.
[Info  - 6:50:36 PM] ESLint library loaded from: C:\Users\gian1\Proyectos\reproduce-vscode-eslint-issue\node_modules\eslint\lib\api.js
[Error - 6:51:00 PM] An unexpected error occurred:
[Error - 6:51:00 PM] Error: ENOENT: no such file or directory, stat 'C:\Users\gian1\Proyectos\reproduce-vscode-eslint-issue\.config\.config\eslint.config.mjs'
[Error - 6:51:00 PM] An unexpected error occurred:
[Error - 6:51:00 PM] Error: ENOENT: no such file or directory, stat 'C:\Users\gian1\Proyectos\reproduce-vscode-eslint-issue\.config\.config\eslint.config.mjs'
```

The error appears when opening `.config\eslint.config.mjs`

### Terminal
```shell
node -v       
v20.17.0


npm run eslint

> reproduce-vscode-eslint-issue@1.0.0 eslint
> eslint . --config .config/eslint.config.mjs


C:\Users\gian1\Proyectos\reproduce-vscode-eslint-issue\src\dummy.mjs
  2:1  warning  Too many blank lines at the beginning of file. Max of 1 allowed  @stylistic/no-multiple-empty-lines
  3:1  warning  Unexpected console statement                                     no-console
  4:1  warning  Too many blank lines at the end of file. Max of 0 allowed        @stylistic/no-multiple-empty-lines

✖ 3 problems (0 errors, 3 warnings)
  0 errors and 2 warnings potentially fixable with the `--fix` option.


```

**Comment by dbaeumer** [comment]
This works for me:

<img width="916" alt="image" src="https://github.com/user-attachments/assets/e2976d81-5f00-4fb7-b938-5396e9d394f8">

**Comment by gian1200** [comment]
Hi, sorry for tha delay.

Please try 2 things:
 - Have also "eslint.config.mjs" file open
 - Open the Output panel:

![image](https://github.com/user-attachments/assets/1926780b-d169-44aa-8957-cbab7e7283db)

Maybe I wasn't so clear in my previous message (https://github.com/microsoft/vscode-eslint/issues/1932#issuecomment-2362105821)

What I found was that even though the config file location is overwritten (in `.vscode/settings.json`) and is found, it is still looking it in additional locations `<project-path>\.config\.config\eslint.config.mjs` (notice how it's twice `.config`).

PS:
Same project with an invalid location:

![image](https://github.com/user-attachments/assets/ea2a3e1a-249a-4774-b0fd-012583139e8d)

**Comment by gian1200** [comment]
I think I've pin point the issue. It happens only for files inside `.config` folder. For any file outside, VSCode-ESLint works fine.

Steps to reproduce:
1. git clone [same reproducer](https://github.com/gian1200/reproduce-vscode-eslint-issue)
2. `cd reproduce-vscode-eslint-issue` and `npm install` inside the root folder
3. `code .` --> open the project on VSCode and make sure vscode-eslint is enabled (wait for everything to load). If not loaded, a VSCode reload might be necessary.
4. Open files for testing:
    - `src/dummy.mjs`: It is flagged correctly by VSCode-ESLint
    - `.config/anotherDummyFile.mjs`: **This file is not flagged at all (see Output for error)**
5. Execute `npm run eslint`: both files are correctly flagged
